pyproject.toml

```diff
@@ -192,3 +190,43 @@ ignore_errors = true
 
 [tool.coverage.html]
 directory = "htmlcov"
+
+[tool.vendoring]
+destination = "lib/spack/external/_vendoring"
+requirements = "lib/spack/external/vendor.txt"
+namespace = ""
+
+protected-files = ["__init__.py", "README.rst", "vendor.txt"]
+patches-dir = "lib/spack/external/patches"
+
+[tool.vendoring.transformations]
+substitute = [
+]
+drop = [
+  # contains unnecessary scripts
+  "bin/",
+  # interpreter and OS specific msgpack libs
+  "msgpack/*.so",
+  # unneeded parts of setuptools
+  "easy_install.py",
+  "setuptools",
+  "pkg_resources/_vendor/",
+  "pkg_resources/extern/",
+  # trim vendored pygments styles and lexers
+  "pygments/styles/[!_]*.py",
+  '^pygments/lexers/(?!python|__init__|_mapping).*\.py$',
```

Comment:
Random comment: I have thought a few times that it would be nice to vendor pygments so we could colorize YAML and JSON output.

Reply:
The only reason pip trims this is for distribution sizes.